GameDataParser/Parsers/SkillParser.cs

```diff
@@ -44,69 +43,57 @@ public static List<SkillMetadata> Parse(MemoryMappedFile m2dFile, IEnumerable<Pa
                     }
                     XmlNode kinds = node.SelectSingleNode("kinds");
                     metadata.Type = kinds.Attributes["type"].Value != null ? int.Parse(kinds.Attributes["type"].Value) : 0;
-                    metadata.SubType = kinds.Attributes["subType"].Value != null ? int.Parse(kinds.Attributes["subType"].Value) : 0;
-                    metadata.RangeType = kinds.Attributes["rangeType"].Value != null ? int.Parse(kinds.Attributes["rangeType"].Value) : 0;
-                    metadata.Element = kinds.Attributes["element"].Value != null ? int.Parse(kinds.Attributes["element"].Value) : 0;
+                    if (kinds.Attributes.GetNamedItem("subType") != null)
+                    {
+                        // Get Rank
+                        metadata.SubType = kinds.Attributes["subType"].Value != null ? int.Parse(kinds.Attributes["subType"].Value) : 0;
+                    }
+                    XmlNode stateAttr = node.SelectSingleNode("stateAttr");
+                    if (stateAttr.Attributes.GetNamedItem("useDefaultSkill") != null)
+                    {
+                        // Get Rank
+                        metadata.DefaultSkill = stateAttr.Attributes["useDefaultSkill"].Value != null ? byte.Parse(stateAttr.Attributes["useDefaultSkill"].Value) : 0;
+                    }
                 }
 
                 XmlNodeList levels = document.SelectNodes("/ms2/level");
                 foreach (XmlNode node in levels)
                 {
-                    int level = node.Attributes["value"].Value != null ? int.Parse(node.Attributes["value"].Value) : 1;
-                    XmlNode consume = node.SelectSingleNode("consume/stat");
```

Comment:
are we still using xml? I see them all deleted

Reply:
Yes, they are being deleted, however, we are parsing skill from files and using it on a metadata file, the same way as `map` or `item`